### DISCUSSION TITLE
Periodic B.C. for CFD-DEM

### DISCUSSION THREAD

**Discussion by JianhuaQin** [chaos-polymtl/lethe#663] [Q&A] [answered]
Hi All,

I am new to Lethe, it seems to be a nice tool for me.
Currently, I want to simulate unresolved CFD-DEM with periodic boundary conditions.
Two questions are bothering me right now.
1. Since I haven't seen any related tutorials about this, is it possible to  use periodic B.C. for unresolved CFD-DEM?
2. If possible, how should I set the boundary conditions? (I tried to use set DEM boundary conditions but is doesn't work)

Best,
Jianhua

**Comment by blaisb** [comment, marked as answer]
Dear @JianhuaQin ,
Periodic boundary conditions already exist on the CFD side for a long time now. Periodic boundary condition are  supported on the DEM side. However, we only support periodic boundary conditions along a single direction (e.g. only the walls on the x direction) for example. There are no examples for this feature yet, but @acdaigneault  will be able to share a zip file here with a basic example where there is a set of periodic boundary conditions. :)

**Reply by JianhuaQin** [reply]
Thanks for the answer, hope the periodic boundary condition will come soon!

**Reply by blaisb** [reply]
There are actually already there :). Sorry, I made a typo above. Periodic BCs are supported on the DEM side, but just along a direction. An example will come in soon :).

**Comment by acdaigneault** [comment]
Hello @JianhuaQin I'm currently adding an example to use periodic boundary conditions in DEM. I will send you files for a case using unresolved CFD-DEM with periodic boundary conditions very soon, just want to make sure to send all the needed files.

**Reply by acdaigneault** [reply]
Here it is, tell me if something is wrong.
[pneumatic_conveying_dilute.zip](https://github.com/lethe-cfd/lethe/files/10219326/pneumatic_conveying_dilute.zip)

**Reply by acdaigneault** [reply]
Hello @JianhuaQin, just letting you know that the example of PBC used on the rectangular hopper with DEM is medged :)

**Reply by JianhuaQin** [reply]
@acdaigneault Thanks for sharing the files!

**Reply by JianhuaQin** [reply]
> Here it is, tell me if something is wrong. [pneumatic_conveying_dilute.zip](https://github.com/lethe-cfd/lethe/files/10219326/pneumatic_conveying_dilute.zip)

@acdaigneault @blaisb This example works! Hope we can have periodic B.C. in multiple directions for DEM simulations soon, that's related to my problem of interest.
